sqlglot/parser.py

```diff
@@ -1810,7 +1810,7 @@ def _parse_property_before(self) -> t.Optional[exp.Expression]:
     def _parse_wrapped_properties(self) -> t.List[exp.Expression]:
         return self._parse_wrapped_csv(self._parse_property)
 
-    def _parse_property(self) -> t.Optional[exp.Expression]:
+    def _parse_property(self, parse_as_var: bool = False) -> t.Optional[exp.Expression]:
```

Comment:
when would parse_as_var be False?

Reply:
`parse_as_var` forces `_parse_property` to parse both LHS and RHS as vars if applicable which is our wanted behavior when parsing the copy options. 

However, we don't want to alter the existing semantics of `_parse_property` which originally parsed `this = _parse_column()` and `value = _parse_bitwise()`, so this defaults to False in all other cases except `parse_copy`

Reply:
I think this code introduced [changed here](https://github.com/tobymao/sqlglot/pull/2296) to allow for identifiers as keys

Reply:
they are identifiers though or dots, not columns

Reply:
so maybe we never need parse_property to return a column?

Reply:
IIRC I'd originally added `_parse_colum()` in the property parser to handle things like `delta.dataSkippingNumIndexedCols = 1` (following the linked [issue](https://github.com/tobymao/sqlglot/issues/2289)).

What we could do instead to improve the AST is to keep the `_parse_column()` call, but call `to_dot()` on the parsed expression, if it's a `Column` instance.

Reply:
Right @georgesittas this was already baked in the return statement, i.e we can have an `exp.Column` but it will always be transformed to `exp.Dot` before constructing the `exp.Property`

Reply:
there doesn't need to be a boolean, we should just always remove columns imo

Reply:
Ah we already did that, lol. Good :)

Reply:
In my latest commit I removed the boolean arg altogether and instead always transform `exp.Column` to `exp.Var`, except the dotted identifier case of Databricks